### PR TITLE
[ZEPPELIN-1279] Zeppelin with CDH5.x docker document.

### PR DESCRIPTION
**Pull request by astroshim**
### What is this PR for?
This PR is for the documentation of running zeppelin with CDH docker environment.
and This PR is the part of https://issues.apache.org/jira/browse/ZEPPELIN-1198.

Tested CDH5.7 on ubuntu.

### What type of PR is it?
Documentation

### What is the Jira issue?
https://issues.apache.org/jira/browse/ZEPPELIN-1281

### Questions:
* Does the licenses files need update? no
* Is there breaking changes for older versions? no
* Does this needs documentation? no

### CODE REVIEW

File: docs/install/spark_cluster_mode.md [hunk 1]
```diff
@@ -202,3 +202,65 @@ After running a single paragraph with Spark interpreter in Zeppelin, browse `htt
 
 <img src="../assets/themes/zeppelin/img/docs-img/mesos_frameworks.png" />
 
+
+## Zeppelin on CDH
+Cloudera officially provide docker container so we can easily build CDH docker environment following the [link](https://www.cloudera.com/documentation/enterprise/5-6-x/topics/quickstart_docker_container.html).
```
Comment (felixcheung) [inline docs/install/spark_cluster_mode.md]: link to https://www.cloudera.com/documentation/enterprise/latest.html
the latest CDH release is 5.8

File: docs/install/spark_cluster_mode.md [hunk 2]
```diff
@@ -202,3 +202,65 @@ After running a single paragraph with Spark interpreter in Zeppelin, browse `htt
...
+```
+
+`HADOOP_CONF_DIR`(Hadoop configuration path) is defined in `/scripts/docker/spark-cluster-managers/cdh/hdfs_conf`.
+
+Don't forget to set Spark `master` as `yarn-client` in Zeppelin **Interpreters** setting page like below.
+
+<img src="../assets/themes/zeppelin/img/docs-img/zeppelin_yarn_conf.png" />
+
+### 4. Run Zeppelin with Spark interpreter
+After running a single paragraph with Spark interpreter in Zeppelin, browse `http://<hostname>:8088/cluster/apps` and check Zeppelin application is running well or not.
```
Comment (felixcheung) [inline docs/install/spark_cluster_mode.md]: should this explain to check zeppelin also?

**Comment by AhyoungRyu** [comment]
@astroshim Thanks for your contribution! This can be really helpful to `CDH` & `Spark` & `Zeppelin` users.



How about separating `Zeppelin on Spark Cluster Mode` page with `Zeppelin on CDH` instead of putting this contents in one page?  Then you need to create `docs/install/cdh.md` maybe.
I think it's more makes to have only `Spark Cluster Mode` contents in `docs/install/spark_cluster_mode.md`.

Or there is an another option. If you think the contents in `Zeppelin on CDH` is too short to have a separated page, you can just modify the title("Apache Zeppelin on Spark Cluster Mode") and the contents of "Overview" in `spark_cluster_mode.md`. And maybe the file name needs to be changed accordingly :)

What do you think?

File: docs/install/spark_cluster_mode.md (hunk 2 above)
Comment (astroshim) [inline docs/install/spark_cluster_mode.md]: Okay. Let me add that. Thanks.

File: docs/install/spark_cluster_mode.md (hunk 1 above)
Comment (astroshim) [inline docs/install/spark_cluster_mode.md]: Thank you for properly pointing out.

**Comment by astroshim** [comment]
@AhyoungRyu Thank you for your suggestion. I think creating new page for `Zeppelin on CDH` is better, what do you think?

**Comment by AhyoungRyu** [comment]
@astroshim Yeah that would be better than all contents are together in one page actually. Thanks for your consideration :D

File: docs/install/cdh.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+### 4. Configure Spark interpreter in Zeppelin
+Set following configurations to `conf/zeppelin-env.sh`.
+
+```
+export MASTER=yarn-client
+export HADOOP_CONF_DIR=[your_hadoop_conf_path]
+export SPARK_HOME=[your_spark_home_path]
+```
+
+`HADOOP_CONF_DIR`(Hadoop configuration path) is defined in `/scripts/docker/spark-cluster-managers/cdh/hdfs_conf`.
```
Comment (AhyoungRyu) [inline docs/install/cdh.md]: @astroshim Do you mean `/scripts/docker/spark-cluster-managers/cdh/*.xml`?

**Comment by AhyoungRyu** [comment]
@astroshim To make CI happy, you might need to exclude `scripts` dir  from the `rat-plugin`. Please add this single line`<exclude>scripts/**</exclude>` to [here](https://github.com/apache/zeppelin/blob/master/pom.xml#L748).

File: docs/install/cdh.md (hunk 3 above)
Comment (astroshim) [inline docs/install/cdh.md]: The configuration for cdh is in the ``/scripts/docker/spark-cluster-managers/cdh/hdfs_conf/` like yarn mode.

Comment (astroshim) [inline docs/install/cdh.md]: Sorry, I made `/scripts/docker/spark-cluster-managers/cdh/hdfs_conf/` and move configurations to that path.

Comment (AhyoungRyu) [inline docs/install/cdh.md]: @astroshim Thanks for your prompt handling :D

**Comment by AhyoungRyu** [comment]
@astroshim still CI is not happy but it seems irrelevant to this change. As below result says, there is a test failure in `zeppelin-display`. 

```
36mTests: succeeded 17, failed 1, canceled 0, ignored 0, pending 0[0m
[31m*** 1 TEST FAILED ***[0m
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Zeppelin ........................................... SUCCESS [  4.051 s]
[INFO] Zeppelin: Interpreter .............................. SUCCESS [02:13 min]
[INFO] Zeppelin: Zengine .................................. SUCCESS [01:53 min]
[INFO] Zeppelin: Display system apis ...................... FAILURE [ 16.048 s]
[INFO] Zeppelin: Spark dependencies ....................... SKIPPED
[INFO] Zeppelin: Spark .................................... SKIPPED
[INFO] Zeppelin: Markdown interpreter ..................... SKIPPED
[INFO] Zeppelin: Angular interpreter ...................... SKIPPED
[INFO] Zeppelin: Shell interpreter ........................ SKIPPED
[INFO] Zeppelin: Livy interpreter ......................... SKIPPED
[INFO] Zeppelin: HBase interpreter ........................ SKIPPED
[INFO] Zeppelin: PostgreSQL interpreter ................... SKIPPED
[INFO] Zeppelin: JDBC interpreter ......................... SKIPPED
[INFO] Zeppelin: File System Interpreters ................. SKIPPED
[INFO] Zeppelin: Flink .................................... SKIPPED
[INFO] Zeppelin: Apache Ignite interpreter ................ SKIPPED
[INFO] Zeppelin: Kylin interpreter ........................ SKIPPED
[INFO] Zeppelin: Python interpreter ....................... SKIPPED
[INFO] Zeppelin: Lens interpreter ......................... SKIPPED
[INFO] Zeppelin: Apache Cassandra interpreter ............. SKIPPED
[INFO] Zeppelin: Elasticsearch interpreter ................ SKIPPED
[INFO] Zeppelin: BigQuery interpreter ..................... SKIPPED
[INFO] Zeppelin: Alluxio interpreter ...................... SKIPPED
[INFO] Zeppelin: web Application .......................... SKIPPED
[INFO] Zeppelin: Server ................................... SKIPPED
[INFO] Zeppelin: Packaging distribution ................... SKIPPED
[INFO] Zeppelin: R Interpreter ............................ SKIPPED
[INFO] Zeppelin: Scalding interpreter ..................... SKIPPED
[INFO] Zeppelin: Examples ................................. SKIPPED
[INFO] Zeppelin: Example application - Clock .............. SKIPPED
[INFO] Zeppelin: Example application - Horizontal Bar chart SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 04:28 min
[INFO] Finished at: 2016-09-27T02:59:19+00:00
[INFO] Final Memory: 33M/421M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.scalatest:scalatest-maven-plugin:1.0:test (test) on project zeppelin-display_2.11: There are test failures -> [Help 1]
[ERROR] 
``` 

Looks good 👍

**Comment by astroshim** [comment]
@AhyoungRyu Thank you so much for taking care of this!!

**Comment by AhyoungRyu** [comment]
@astroshim Appreciate for your effort! Merge into master if there are no more discussions :)

**Pull request closed without merging**